src/items/enumerations.md

```diff
@@ -61,6 +62,27 @@ integer associated to it that is used to determine which variant it holds. An
 opaque reference to this discriminant can be obtained with the
 [`mem::discriminant`] function.
 
+Variant constructors are similar to [struct] definitions, and can be referenced by a path from the enumeration name, including in [use declarations].
+The constructors are defined in both the [type namespace] and [value namespace] within the enumeration.
```

Comment:
Can you clarify something here?  The tuple-struct constructor in the value namespace doesn't currently have much of a purpose, does it?  At least, I can't think of a way it could be used. The only thing I noticed is that you are not allowed to shadow it, as in:

```rust
enum E {
    V1 {f1: i32}
}

fn foo() {
    use E::*;
    let V1 = 123; // ERROR: let bindings cannot shadow struct variants
}
```

Reply:
The value namespace component for braced variants was removed in https://github.com/rust-lang/rust/pull/103578.

Reply:
Thanks!